zml/aio/torch.zig

```diff
@@ -314,79 +314,76 @@ pub const PickleData = struct {
             .pers_id => |pers_id| try self.parseValue(allocator, store, prefix, pers_id.ref),
             .seq => |*seq| switch (seq[0]) {
                 .list, .tuple, .set, .frozen_set => {
-                    const elemCheck = struct {
-                        fn call(comptime T: ValueType) fn (v: Value) bool {
-                            return struct {
-                                fn call(val: Value) bool {
-                                    return val == T;
+                    if (seq[1].len == 0) return;
+                    var valid_slice = true;
+                    switch (seq[1][0]) {
+                        inline .int, .float, .bool => |val0, tag| {
+                            var values: std.ArrayListUnmanaged(switch (tag) {
+                                .int => i64,
+                                .float => f64,
+                                .bool => bool,
+                                else => unreachable,
```

Comment:
can you extract the switch to its own line, to clearly separate what is comptime from what is not ?